src/frontend/src/components/top-bar.js

```diff
@@ -46,7 +46,9 @@ class TopBar extends PureComponent<Props, State> {
     const query = {
       zipcode: this.props.neighborhood.properties.id,
       budget: this.props.estMaxRent,
-      rooms: this.props.userProfile.rooms,
+      rooms: this.props.userProfile.hasVoucher
```

Comment:
There are enough of these ternaries that it might be worth adding a utility function so that the logic only exists in one place. However, it feels like a borderline case to me; the logic is really simple, and I suspect that there would be a readability cost to adding a function because it would likely end up being called something like `getProfileRooms()`, which could obscure what's going on. I also don't think it would be too hard to grep for all the places we use `hasVoucher` and change them if we need to.

Anyway, that's all to say that if you haven't thought about putting these into a function (or two, because there is the budget to handle as well) it'd be worth thinking about for a few minutes, but I don't have a strong preference either way (and in fact I may have talked myself around to leaning toward keeping things as-is).

Reply:
Thanks for reviewing and for pointing this out--it hadn't occurred to me to put them in functions but I wanted to give it a try because it did feel cleaner. However, after trying it, I found that attempting to apply the same rules wherever a ternary is currently being used (if the user has no voucher or budget set, use neighborhood max rent; if the user has no rooms set, use 0) broke the external links like Zillow that were smart enough to use null values. As a result, I'm inclined to keep it as is, because it seems like I would need to add more logic to each query anyway to make it behave the way it currently does. So I'm going to leave it and merge now but, as you said, we can always replace them if we decide that makes more sense later.